File: C Programming/Quadratic Formula/Quadratic Formula.c
#include <stdio.h>
#include <math.h>

double getSquareRoot(double num1);
double getSquare(double num1);
double getAbsValue(double num1);

int main(int argc, char const *argv[])
{
	double a, b, c, root1, root2, root, discriminantFormula;

	printf("Enter the value for a, b, and c: \n");

	scanf("%lf %lf %lf", &a, &b, &c);

	discriminantFormula = (getSquare(b)) - (4.0 * a * c);

	printf("%lf\n", discriminantFormula);

	if (discriminantFormula < 0.0)
	{
		printf("There are two roots these are: \n");
		root1 = -b + ((getSquareRoot(discriminantFormula)) / (2.0 * a));
		root2 = -b - ((getSquareRoot(discriminantFormula)) / (2.0 * a));
		printf("First root:\t%lf\nSecond root:\t%lf\n", root1, root2);
	}
 else if (discriminantFormula > 0.0) {
  root1 = -b + ((getSquareRoot(discriminantFormula)) / (2.0 * a));
		root2 = -b - ((getSquareRoot(discriminantFormula)) / (2.0 * a));
  printf("The roots are imaginary!\nThese are %.2lfi and %.2lfi\n");
	else{
		printf("There is only one root: \n");
		root = -b + ((getSquareRoot(discriminantFormula)) / (2.0 * a));
		printf("The root is:\t%lf\n", root);
	}
	return 0;
}

double getSquareRoot(double num1){
	return sqrt(num1);
}

double getSquare(double num1){
	return num1 * num1;
}

double getAbsValue(double num1){
 return abs(num1);
}

File: Python Programming/Quadratic Formula/Quadratic Formula.py
import math


a = float(input("Enter the value for a: "))
b = float(input("Enter the value for b: "))
c = float(input("Enter the value for c: "))

discriminantFormula = (b**2) - (4 * a * c)

if discriminantFormula > 0:
    root1 = -b + (math.sqrt(discriminantFormula) / (2 * a))
    root2 = -b - (math.sqrt(discriminantFormula) / (2 * a))
    print(f"There are two roots these are {root1} and {root2}.")

elif discriminantFormula < 0:
    discriminantFormula = abs(discriminantFormula)
    root1 = -b + (math.sqrt(discriminantFormula) / (2 * a))
    root2 = -b - (math.sqrt(discriminantFormula) / (2 * a))
    print(f"The roots are imaginary!\nThese are {root1}i and {root2}i.")

else:
    root = -b + (math.sqrt(discriminantFormula) / (2 * a))
    print(f"There is only one root which is {root}")
